.github/workflows/test_cpu.yml

```
@@ -0,0 +1,79 @@
+name: Run tests on CPU
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    types: [open, reopened, synchronize, review_requested, ready_for_review]
+  schedule:
+    - cron: "0 4 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests:
```

```suggestion
  tests-cpu:
```
i hope soon we can support gpu too